### DISCUSSION TITLE
Coroot doesn't recognize coroot-node-agent

### DISCUSSION THREAD

**Discussion by veer-pradhan** [coroot/coroot#139] [Q&A]
Hi All,

I am developing a POC for distributed tracing using coroot-node-agent. I am running the coroot ui and node-agent in docker containers:

`sudo docker run -d --name coroot-node-agent -p 8081:8081 --privileged --pid host -v /sys/kernel/debug:/sys/kernel/debug:rw -v /sys/fs/cgroup:/host/sys/fs/cgroup:ro ghcr.io/coroot/coroot-node-agent --cgroupfs-root=/host/sys/fs/cgroup`
`sudo docker run -d --name coroot -p 8080:8080 ghcr.io/coroot/coroot `

It does not recognize the coroot-node-agent is also being run. Any help would be appreciated!
<img width="1591" alt="Screenshot 2024-01-02 at 1 09 51 PM" src="https://github.com/coroot/coroot/assets/34077108/974d7351-fcc9-4525-94f4-cb8b8ae8eb83">

**Comment by apetruhin** [comment]
@veer-pradhan
It seems that Coroot is not detecting metrics from the node-agent.

1. Check the `/metrics` endpoint of the node-agent:
```
curl http://localhost:8081/metrics
```

2. Ensure that the Prometheus configuration includes the correct target for the node-agent. Look for the `scrape_configs` section in your Prometheus configuration file. Sample configuration:
```yaml
scrape_configs:
  - job_name: 'coroot-node-agent'
    static_configs:
      - targets: ['localhost:8081']
```

3. Access the Prometheus Targets page (usually `/targets`) and check if the node-agent target appears as "UP."

**Reply by apetruhin** [reply]
You should fix the port mapping to `-p 8081:80` since the agent listens on port 80 by default.

**Comment by T100D** [comment]
We have had similar problems. In our case localhost should be changed in the host-ip address in the target address.
